### PR TITLE
About Dialog - Report issues URL added to branding

### PR DESCRIPTION
**Pull request by yodem**
So far, both in RHV and oVirt, the -"About" window dialog included a reference to github to report issues. However In RHV the customers supposed to report to access.redhat.com in case of a problem.
Now the report reference depends on what brand do you use.

Reports1

Fixed  https://bugzilla.redhat.com/show_bug.cgi?id=1724959

### CODE REVIEW

File: src/intl/messages.js [hunk 1]
```diff
@@ -221,7 +221,7 @@ export const messages: { [messageId: string]: MessageType } = {
     description: 'Being hibernated. One of states of a virtual machine. Other are e.g. Up, Down, Powering-Up',
   },
   enum_VmStatus_suspended: {
-    message: 'Asleep',
+    message: 'Suspended',
```
Comment (bond95) [inline src/intl/messages.js]: I guess this commit is from other PR

File: src/components/VmDetails/CardEditButton.js [hunk 2]
```diff
@@ -44,9 +46,11 @@ class CardEditButton extends React.Component {
     const myClick = editEnabled ? noop : this.enableEditHandler
 
     return (
-      <a title={tooltip} onClick={(e) => { e.preventDefault(); myClick() }} className={classes} id={id}>
-        <Icon type='pf' name='edit' />
-      </a>
+      <OverlayTrigger overlay={<Tooltip id={`${id}-edit-tooltip`}>{tooltip}</Tooltip>} placement='top' id={id} onClick={(e) => { e.preventDefault(); myClick() }} >
```
Comment (bond95) [inline src/components/VmDetails/CardEditButton.js]: This one too.

**Comment by yodem** [comment]
I working on in, i figured out what is my mistake and i'll fix it :)

File: branding/fixed-strings.json [hunk 3]
```diff
@@ -2,5 +2,8 @@
   "BRAND_NAME": "oVirt",
   "LEGAL_INFO": "Released under",
   "LEGAL_INFO_LINK_TEXT": "Apache License 2.0",
-  "LEGAL_INFO_LINK_URL": "https://github.com/oVirt/ovirt-web-ui/blob/master/LICENSE"
+  "LEGAL_INFO_LINK_URL": "https://github.com/oVirt/ovirt-web-ui/blob/master/LICENSE",
+  "GITHUB_ISSUES_TRACKER_URL": "https://github.com/oVirt/ovirt-web-ui/issues",
```
Comment (bond95) [inline branding/fixed-strings.json]: I think will be enough leave attributes with name `ISSUES_TRACKER_URL` and `ISSUES_TRACKER_NAME`. And then use it in `About` component.

File: src/components/About.js [hunk 4]
```diff
@@ -57,7 +57,7 @@ class AboutDialog extends React.Component {
                 <ul className='list-unstyled'>
                   <li id={`${idPrefix}-version`}>Version <strong id={`${idPrefix}-version-value`}>{Product.version}-{Product.release}</strong></li>
                   <li id={`${idPrefix}-apiversion`}>{fixedStrings.BRAND_NAME} API Version <strong id={`${idPrefix}-apiversion-value`}>{apiVersion}</strong></li>
-                  <li id={`${idPrefix}-issues`}>Please report issues on <strong><a href='https://github.com/oVirt/ovirt-web-ui/issues' target='_blank' id={`${idPrefix}-issues-link`}>GitHub Issue Tracker</a></strong></li>
+                  <li id={`${idPrefix}-issues`}>Please report issues on <strong><a href={fixedStrings.BRAND_NAME === 'oVirt' ? fixedStrings.GITHUB_ISSUES_TRACKER_URL : fixedStrings.REDHAT_CUSTOMER_PORTAL_URL} target='_blank' id={`${idPrefix}-issues-link`}>{fixedStrings.BRAND_NAME === 'oVirt' ? 'GitHub Issue Tracker' : 'Redhat Customer Portal'}</a></strong></li>
```
Comment (bond95) [inline src/components/About.js]: Like this, for example:
```JavaScript
<li id={`${idPrefix}-issues`}>Please report issues on <strong><a href={fixedStrings.ISSUES_TRACKER_URL} target='_blank' id={`${idPrefix}-issues-link`}>{fixedStrings.ISSUES_TRACKER_NAME}</a></strong></li>
```

Comment (yodem) [reply src/components/About.js]: ohhh and then it will automatically change because RHV fixedStrings will override oVirt fixedStrings?

File: branding/fixed-strings.json (hunk 3 above)
Comment (bond95) [reply branding/fixed-strings.json]: Please, update `fixed-strings.json`, I mean, remove `REDHAT_CUSTOMER_PORTAL_URL` and `GITHUB_ISSUES_TRACKER_URL` and add `ISSUES_TRACKER_URL`, `ISSUES_TRACKER_TEXT`

File: src/components/About.js (hunk 4 above)
Comment (bond95) [reply src/components/About.js]: RHV fixedStrings replace oVirt

Comment (yodem) [reply src/components/About.js]: oh I forgot to add fixedstrings to the commit...

**Comment by sgratch** [comment]
Can you please add a screenshot of how it looks like in RHV now?

**Comment by michalskrivanek** [comment]
then there should be a condition too in case they are not defined. we can't easily add a package dependency

**Comment by michalskrivanek** [comment]
...though, if it doesn't explode the whole page it's probably fine. it's not too important

**Comment by yodem** [comment]
Is there anything else to do about this issue?

**Comment by michalskrivanek** [comment]
well, to check if it does or does not explode when not defined:)

**Comment by yodem** [comment]
checks added :)

File: src/components/About.js [hunk 5]
```diff
@@ -57,7 +57,9 @@ class AboutDialog extends React.Component {
                 <ul className='list-unstyled'>
                   <li id={`${idPrefix}-version`}>Version <strong id={`${idPrefix}-version-value`}>{Product.version}-{Product.release}</strong></li>
                   <li id={`${idPrefix}-apiversion`}>{fixedStrings.BRAND_NAME} API Version <strong id={`${idPrefix}-apiversion-value`}>{apiVersion}</strong></li>
-                  <li id={`${idPrefix}-issues`}>Please report issues on <strong><a href='https://github.com/oVirt/ovirt-web-ui/issues' target='_blank' id={`${idPrefix}-issues-link`}>GitHub Issue Tracker</a></strong></li>
+                  {fixedStrings.ISSUES_TRACKER_URL && fixedStrings.ISSUES_TRACKER_TEXT &&
+                    <li id={`${idPrefix}-issues`}>Please report issues on <strong><a href={fixedStrings.ISSUES_TRACKER_URL} target='_blank' id={`${idPrefix}-issues-link`}>{fixedStrings.ISSUES_TRACKER_TEXT}</a></strong></li>
```
Comment (sjd78) [inline src/components/About.js]: The "Please report issues on {link}" text should probably be externalized to `src/intl/messages.js`

File: src/GlobalErrorBoundary.js [hunk 6]
```diff
@@ -25,9 +25,9 @@ class GlobalErrorBoundary extends React.Component {
 
   render () {
     const title = msg.globalErrorBoundaryTitle()
-    const descr = msg.globalErrorBoundaryDescription({
-      bugUrl: `<a href='https://github.com/oVirt/ovirt-web-ui/issues'>${msg.gitHub()}</a>`,
-    })
+    const descr = branding.fixedStrings.ISSUES_TRACKER_URL ? msg.globalErrorBoundaryDescription({
```
Comment (sjd78) [inline src/GlobalErrorBoundary.js]: This is a bit awkward.
Like in the About box, if the tracker isn't in branding, the "report issues" bit should be hidden instead of an empty link.

Maybe have a `const showDescription = branding.fixedString.ISSUES_TRACKER URL && .....ISSUES_TRACKER_TEXT` and then the `descr` and the html would be wrapped by that constant.

File: src/components/About.js (hunk 5 above)
Comment (yodem) [reply src/components/About.js]: i agree on the  "Please report issues on " but i dont think that you need to add the link. The link has <a>,<strong> tags and id which the "Please report issues on " dont have. and to insert it all as a json object and to pass it through a function make the code more complex instead of leave it as is, which is readable and clear

File: src/GlobalErrorBoundary.js (hunk 6 above)
Comment (yodem) [reply src/GlobalErrorBoundary.js]: the descr need to be wrapped as well? i cant wrap him inside the return section.
I can check if showDescription exists before creating the descr, but it needs to be before the return statement.
right?

**Comment by bond95** [comment]
I think, it will be nice to add default values in case if `ISSUES_TRACKER_URL` and `ISSUES_TRACKER_TEXT` are not defined.
@yodem @sjd78 @michalskrivanek WDYT?

**Comment by yodem** [comment]
> I think, it will be nice to add default values in case if `ISSUES_TRACKER_URL` and `ISSUES_TRACKER_TEXT` are not defined.
> @yodem @sjd78 @michalskrivanek WDYT?

I thought about it, but what would be the default value?

**Comment by bond95** [comment]
> I thought about it, but what would be the default value?

As it was before, ovirt-web-ui github.

**Comment by yodem** [comment]
> As it was before, ovirt-web-ui github.

ok

File: src/GlobalErrorBoundary.js [hunk 7]
```diff
@@ -24,9 +24,11 @@ class GlobalErrorBoundary extends React.Component {
   }
 
   render () {
+    const trackText = branding.fixedStrings.ISSUES_TRACKER_TEXT ? branding.fixedStrings.ISSUES_TRACKER_TEXT : 'https://github.com/oVirt/ovirt-web-ui/issues'
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/GlobalErrorBoundary.js]: You have the default value as the URL instead of the text.

File: src/GlobalErrorBoundary.js [hunk 8]
```diff
@@ -24,9 +24,11 @@ class GlobalErrorBoundary extends React.Component {
...
+    const trackUrl = branding.fixedStrings.ISSUES_TRACKER_URL ? branding.fixedStrings.ISSUES_TRACKER_URL : 'Github Issue Tracker'
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/GlobalErrorBoundary.js]: You have the default value as the text instead of the URL.

File: src/components/About.js [hunk 9]
```diff
@@ -37,6 +37,8 @@ class AboutDialog extends React.Component {
 
     const { oVirtApiVersion } = this.props
     const idPrefix = `about`
+    const trackText = fixedStrings.ISSUES_TRACKER_TEXT ? fixedStrings.ISSUES_TRACKER_TEXT : 'https://github.com/oVirt/ovirt-web-ui/issues'
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/components/About.js]: You have the default value as the URL instead of the text.

File: src/components/About.js [hunk 10]
```diff
@@ -37,6 +37,8 @@ class AboutDialog extends React.Component {
...
+    const trackUrl = fixedStrings.ISSUES_TRACKER_URL ? fixedStrings.ISSUES_TRACKER_URL : 'Github Issue Tracker'
```
[the remark this answers is not in the document]
Comment (sjd78) [reply src/components/About.js]: You have the default value as the text instead of the URL.

**Comment by yodem** [comment]
> Thanks for making the changes to the translation key and text.
> 
> @yodem - Any particular reason to avoid the style change away from the ternary to just the `||` for the default `trackText` and `trackUrl` constants? The form `const a = data.value || 'constant'` is quite common for default values when a variable is not set.

Good to know, I'll change it.

**Comment by sgratch** [comment]
ci build please

**Comment by sjd78** [comment]
@yodem - There are lint errors causing the tests (and build) to fail: https://travis-ci.org/oVirt/ovirt-web-ui/builds/607522440#L275

**Comment by sgratch** [comment]
@yodem moving the ON_QA label since QE can't test till build is passed

**Comment by sgratch** [comment]
@leistnerova @isaranova Do you want to verify this fix before merging or do you prefer to test it only after a d/s build is ready?

File: src/components/NoLogin/index.js [hunk 11]
```diff
@@ -1,7 +1,6 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-
-import { Button } from '@patternfly/react-core'
+import { Button } from 'patternfly-react'
```
Comment (sjd78) [inline src/components/NoLogin/index.js]: This is an unrelated change that should be removed.

File: src/index.js [hunk 12]
```diff
@@ -13,7 +13,6 @@ import { type Task } from 'redux-saga'
 
 import '_/logger' // initialize our console logging overlay
 
-import '@patternfly/react-core/dist/styles/base.css'
```
Comment (sjd78) [inline src/index.js]: This is an unrelated change that should be removed.

**Comment by leistnerova** [comment]
> @leistnerova @isaranova Do you want to verify this fix before merging or do you prefer to test it only after a d/s build is ready?
We will test it on d/s so it can be merged now.
